### ISSUE TITLE
New lint: existing trait method became `unsafe`

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#605]
Major breaking change because existing impls of that trait now need to mark that method `unsafe` as well.

Example: ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=db69136208cdae9377475e0c732dee0d))
```rust
trait Example {
    unsafe fn method();
}

struct MyStruct;

impl Example for MyStruct {
    fn method() {}
}
```

This errors out:
```
error[E0053]: method `method` has an incompatible type for trait
 --> src/lib.rs:8:5
  |
8 |     fn method() {}
  |     ^^^^^^^^^^^ expected unsafe fn, found normal fn
  |
note: type in trait
 --> src/lib.rs:2:5
  |
2 |     unsafe fn method();
  |     ^^^^^^^^^^^^^^^^^^^
  = note: expected signature `unsafe fn()`
             found signature `fn()`

For more information about this error, try `rustc --explain E0053`.
```

Going in the opposite direction is breaking too: #606

**Comment by yottalogical** [comment]
Does #608 close this issue?

**Comment by obi1kenobi** [comment]
Ah yes, good catch -- thanks!
